modules/mapper-extras/src/test/resources/rest-api-spec/test/search-as-you-type/20_highlighting.yml

```diff
@@ -0,0 +1,202 @@
+          highlight:
+            fields:
+              a_field:
+                type: unified
+              a_field._2gram:
+                type: unified
+              a_field._3gram:
+                type: unified
+              a_field._4gram:
+                type: unified
+
+  - match: { hits.total: 1 }
+  - match: { hits.hits.0._source.a_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0._source.text_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0.highlight.a_field: ["quick <em>red</em> fox lazy brown dog"] }
+  - match: { hits.hits.0.highlight.a_field\._2gram: null }
+  - match: { hits.hits.0.highlight.a_field\._3gram: null }
+  - match: { hits.hits.0.highlight.a_field\._4gram: null }
+
+---
+"multi match bool prefix query 2 complete term":
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          query:
+            multi_match:
+              query: "red fox la"
+              type: "bool_prefix"
+              fields: [ "a_field", "a_field._2gram", "a_field._3gram", "a_field._4gram" ]
+          highlight:
+            fields:
+              a_field:
+                type: unified
+              a_field._2gram:
+                type: unified
+              a_field._3gram:
+                type: unified
+              a_field._4gram:
+                type: unified
+
+  - match: { hits.total: 1 }
+  - match: { hits.hits.0._source.a_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0._source.text_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0.highlight.a_field: ["quick <em>red</em> <em>fox</em> lazy brown dog"] }
+  - match: { hits.hits.0.highlight.a_field\._2gram: ["quick <em>red fox</em> lazy brown dog"] }
+  - match: { hits.hits.0.highlight.a_field\._3gram: null }
+  - match: { hits.hits.0.highlight.a_field\._4gram: null }
+
+---
+"multi match bool prefix query 3 complete term":
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          query:
+            multi_match:
+              query: "red fox lazy br"
+              type: "bool_prefix"
+              fields: [ "a_field", "a_field._2gram", "a_field._3gram", "a_field._4gram" ]
+          highlight:
+            fields:
+              a_field:
+                type: unified
+              a_field._2gram:
+                type: unified
+              a_field._3gram:
+                type: unified
+              a_field._4gram:
+                type: unified
+
+  - match: { hits.total: 1 }
+  - match: { hits.hits.0._source.a_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0._source.text_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0.highlight.a_field: ["quick <em>red</em> <em>fox</em> <em>lazy</em> brown dog"] }
+  - match: { hits.hits.0.highlight.a_field\._2gram: ["quick <em>red fox</em><em> lazy</em> brown dog"] }
```

Comment:
This highlighting behavior on the shingle fields smaller than the matching one is a little questionable

Reply:
Highlighting should be done on the root field so I don't think it's an issue.